w3c/ash-nazg
# Fix JS bug “null” related to groups with no participants

(Reported by @sideshowbarker.)

How to reproduce:

1. Go to edit user `yoichio`:
  [`labs.w3.org/hatchery/repo-manager/admin/user/yoichio`](https://labs.w3.org/hatchery/repo-manager/admin/user/yoichio).
1. Try to set their W3C ID (bottom of the page).
1. This error occurs:
  *&ldquo;cannot read property 'href' of null&rdquo;* (top of the page)
  and the ID is *not* set.

My take:

There are groups with zero participants&nbsp;&mdash;&nbsp;like the Web Applications WG [see [in ash-nazg API](https://labs.w3.org/hatchery/repo-manager/api/w3c/group/42538/); see [in member-only pages](https://www.w3.org/2000/09/dbwg/details?group=42538#stats)], to which this user belongs. When the list of participants for that group is retrieved in `/admin/user/yoichio`, `u.href` in [this loop](https://github.com/w3c/ash-nazg/blob/master/application/admin/edit-user.jsx#L81-L85) is evaluated without checking first if there are users at all.

I have patched this API oddity the same way it's done in other parts of the code: by assuming that *&ldquo;sometimes you get a 404, just handle it&rdquo;*.

(Tip: [append `?w=1` to the URL of the &ldquo;files changed&rdquo; view of this PR](https://github.com/w3c/ash-nazg/pull/66/files?w=1) to hide whitespace-only changes in the diff.)
